tests/components/person/test_init.py

```diff
@@ -0,0 +1,160 @@
+        {ATTR_LATITUDE: 10.123456, ATTR_LONGITUDE: 11.123456})
+    await hass.async_block_till_done()
+
+    state = hass.states.get('person.tracked_person')
+    assert state.state == 'not_home'
+    assert state.attributes.get(ATTR_LATITUDE) == 10.12346
+    assert state.attributes.get(ATTR_LONGITUDE) == 11.12346
+    assert state.attributes.get(ATTR_SOURCE) == DEVICE_TRACKER
+    assert state.attributes.get(ATTR_USER_ID) == user_id
+
+
+async def test_setup_two_trackers(hass, hass_owner_user):
+    """Test set up person with two device trackers."""
+    user_id = hass_owner_user.id
+    config = {DOMAIN: {
+        'name': 'tracked person', 'user_id': user_id,
+        'device_trackers': [DEVICE_TRACKER, DEVICE_TRACKER_2]}}
+    assert await async_setup_component(hass, DOMAIN, config)
+
+    state = hass.states.get('person.tracked_person')
+    assert state.state == STATE_UNKNOWN
+    assert state.attributes.get(ATTR_LATITUDE) is None
+    assert state.attributes.get(ATTR_LONGITUDE) is None
+    assert state.attributes.get(ATTR_SOURCE) is None
+    assert state.attributes.get(ATTR_USER_ID) == user_id
+
+    hass.states.async_set(DEVICE_TRACKER, 'home')
+    await hass.async_block_till_done()
+
+    state = hass.states.get('person.tracked_person')
+    assert state.state == 'home'
+    assert state.attributes.get(ATTR_LATITUDE) is None
+    assert state.attributes.get(ATTR_LONGITUDE) is None
+    assert state.attributes.get(ATTR_SOURCE) == DEVICE_TRACKER
+    assert state.attributes.get(ATTR_USER_ID) == user_id
+
+    hass.states.async_set(
+        DEVICE_TRACKER_2, 'not_home',
+        {ATTR_LATITUDE: 12.123456, ATTR_LONGITUDE: 13.123456})
+    await hass.async_block_till_done()
+
+    state = hass.states.get('person.tracked_person')
+    assert state.state == 'not_home'
+    assert state.attributes.get(ATTR_LATITUDE) == 12.12346
+    assert state.attributes.get(ATTR_LONGITUDE) == 13.12346
+    assert state.attributes.get(ATTR_SOURCE) == DEVICE_TRACKER_2
+    assert state.attributes.get(ATTR_USER_ID) == user_id
+
+
+async def test_restore_home_state(hass, hass_owner_user):
+    """Test that the state is restored for a person on startup."""
+    attrs = {
+        ATTR_LATITUDE: 10.12346, ATTR_LONGITUDE: 11.12346,
+        ATTR_SOURCE: DEVICE_TRACKER}
+    state = State('person.tracked_person', 'home', attrs)
+    mock_restore_cache(hass, (state, ))
+    hass.state = CoreState.starting
+    mock_component(hass, 'recorder')
+    user_id = hass_owner_user.id
+    config = {DOMAIN: {
+        'name': 'tracked person', 'user_id': user_id,
+        'device_trackers': DEVICE_TRACKER}}
+    assert await async_setup_component(hass, DOMAIN, config)
+
+    state = hass.states.get('person.tracked_person')
+    assert state.state == 'home'
+    assert state.attributes.get(ATTR_LATITUDE) == 10.12346
+    assert state.attributes.get(ATTR_LONGITUDE) == 11.12346
+    # When restoring state the entity_id of the person will be used as source.
```

Comment:
Is this ok?

Reply:
Yeah I think that should be fine. Worst is that a person temporarily has a wrong state if the user has changed it's entity registry around while HASS was turned off, which seems like an edge case.